services/stats/stats.go

```diff
@@ -133,7 +133,7 @@ func (s *statsdStats) Start(ctx context.Context) {
 		if err != nil {
 			s.state.client.statsd, err = s.getNewStatsdClientWithExpoBackoff(ctx, s.state.conn, s.conf.statsdTagsFormat(), s.conf.statsdDefaultTags())
 			if err != nil {
-				s.conf.enabled = false
+				s.conf.enabled.Store(false)
```

Comment:
What if we keep enabled as a simple bool, by adding a new field for example:

```suggestion
				s.err.Store(fmt.Errorf("stats client: %w", err))
```

Reply:
1. The problem is that `enabled` is checked when skipping metrics as well. In any case, please keep in mind that the real contention with atomic here would happen only if someone is actually doing a write. Luckily we write into `enabled` only once (concurrently).
    * The value is set at the beginning and there is no concurrency.
    * Then in the go routine, after all the backoff retries have failed, we give up and write `false` to `enabled` (only once). 
    * There will be no more writes after that. Everything else is a read and an `atomic.Bool` read is super fast (unlike with a mutex). So I don't have performance concerns because there will be virtually no locking (or very little overhead anyway) considering that 99.99% of the times we only **read** `enabled` (see benchmark below).
2. We don't have this race in the OTEL version so I think that this fix is OK for now.

```go
func BenchmarkBool(b *testing.B) {
	b.Run("bool", func(b *testing.B) {
		var x bool
		for i := 0; i < b.N; i++ {
			if !x {
				go func() {
					x = true
				}()
			}
		}
	})
	b.Run("atomic", func(b *testing.B) {
		var x atomic.Bool
		for i := 0; i < b.N; i++ {
			if !x.Load() {
				go func() {
					x.Store(true)
				}()
			}
		}
	})
}
```

Gives:

```
cpu: 12th Gen Intel(R) Core(TM) i9-12900K
BenchmarkBool
BenchmarkBool/bool
BenchmarkBool/bool-24         	1000000000	         0.2157 ns/op
BenchmarkBool/atomic
BenchmarkBool/atomic-24       	1000000000	         0.2178 ns/op
PASS
```

Reply:
My concern was about using and thus polluting a configuration struct with logic with a state that should belong to `statsdStats`. This is semantic since the conf struct is private, otherwise, it would be strange to ask people to pass atomics to a config.

Also, given that we are going to remove this code latter, I agree that this change is sufficient.